Feat: Redesign Payments Tab

m-sameer opened the pull request:
Fixes #122 

![20181209_140739](https://user-images.githubusercontent.com/28915865/49695131-2223e000-fbbc-11e8-895f-2616b1f8c515.gif)


- Redesign Payments Tab and transfer fragment has been moved to the Finance tab
- I have used a dummy TextView in the ViewPager

Checks:

- [x] Apply the `AndroidStyle.xml` style template to your code in Android Studio

- [x] Run the unit tests with `./gradlew check` to make sure you didn't break anything

- [x] If you have multiple commits please combine them into one commit by squashing them

luckyman20 commented:
@m-sameer Please see the prototypes carefully, you have created blank screens whereas you need to create contents in it too.

ankurs287 commented:
@m-sameer There should be appropriate placeholders when there is no history/invoices/etc.

letelete commented:
It's my bad I didn't mention that in the issue description but would be great if you could add placeholders for the `HISTORY` and `INVOICES` tabs since these might be empty.

(Just like in the Transactions History)

![72f8a9b5-4296-4ee4-8110-8a507251d4d7](https://user-images.githubusercontent.com/18665370/49688093-3bba1e80-fb0d-11e8-95ba-513feee4cf03.png)

m-sameer commented:
@ankurs287 I have added empty states to history and invoice tab but there are no empty states for send and request so I have added the first **TextView** in the screen.

letelete commented:
Btw. I think we could improve code quality  and avoid code repeat by creating `setupStateView(int res, String title, String subtitle)` method in the BaseFragment/Activity

m-sameer commented:
> Btw. I think we could improve code quality and avoid code repeat by creating `setupStateView(int res, String title, String subtitle)` method in the BaseFragment/Activity

@letelete The first parameter of the method, I think, must be the resource id of the drawable resource. We cannot just have the drawable resource, title, and subtitle as the parameter we will also going to need the **ImageView** and two **TextView**s as the parameter. That will be 6 parameters and 6 parameters for a single method are too much as Java doesn't support default parameters so every time all the 6 parameters should be provided which is not much liked by programmers in general.

letelete commented:
Oh yeah, you're right about drawable. I was just quickly checking the class so didn't notice the drawable in the argument.

We don't need views as parameters, just take a look at how toolbar is implemented. It's going to be 3 arguments in the method.

letelete commented:
Btw since Base is an activity/fragment we can still handle getResources.getDrawable(**res**) easily and improve readability by changing the method to the setupStateView(**int res**, String title, String subtitle)

letelete commented:
So the example call could look like this: 

```
private void showErrorState() {
        setupErrorState(R.id.error_drawable, getString(R.string.title),
                getString(R.string.subtitle));
}
```

We're can save so many redundant code repeat by this solution

m-sameer commented:
> We don't need views as parameters, just take a look at how toolbar is implemented. It's going to be 3 arguments in the method.

I think we do need views as parameters, toolbar works because we set the toolbar title in **BaseActivity** by calling `setTitle`. We cannot do the same in case of these views. We will need the view so set its image/text.

letelete commented:
I'm not sure about it, but if its true, we can make use of custom views in this case and it still will reduce code repeats
